lib/puppet/configurer.rb

```diff
@@ -246,6 +246,7 @@ def run_internal(options)
   private :run_internal
 
   def send_report(report)
+    puts report.display_changes if Puppet[:show_changes]
```

Comment:
During PR triage we were talking about whether it makes sense to combine the changes and summary in the `report.summary` method. Puppet already has too many settings, so we are leery about adding new ones.

That said, it looks like the `display_changes` will print one line for each (type,event) tuple, plus the blank between each type, so it could easily add hundreds of lines of output. So I'm okay having it be a separate option.